sklearn/linear_model/tests/test_common.py

```diff
@@ -54,6 +55,11 @@ def test_linear_model_normalize_deprecation_message(
         model.fit(X, y)
     # Filter record in case other unrelated warnings are raised
     unwanted = [r for r in record if r.category != warning_category]
+    if platform.system() == "Darwin" and platform.processor() == "arm":
+        # On macOS with a Apple Silicon M1 processor, we can get a spurious
+        # RuntimeWarning when executing this code. Let us just filter those
+        # warnings out. See: https://github.com/numpy/numpy/issues/18555
+        unwanted = [r for r in unwanted if r.category != RuntimeWarning]
```

Comment:
This test is kind of flaky with `warning_category=None` since it catches everything, when I think it wants to focus on the deprecation warning. Maybe we can define `unwanted` depending on `warning_category`?

```python
if warning_category is None:
    unwanted = [r for r in record if r.category == FutureWarning]
else:
    unwanted = [r for r in record if r.category != warning_category]
```

I think this fix will end up to be temporary, so I am okay with leaving the PR as is.